custom_components/tuya_local/devices/pgst_pw150_alarm.yaml

```diff
@@ -0,0 +1,195 @@
+name: Alarm system
+products:
+  - id: bxtloihrivkxalgn
+    name: PGST PW150 WiFi Home Alarm System
+primary_entity:
+  entity: alarm_control_panel
+  dps:
+    - id: 1
+      type: string
+      name: alarm_state
+      mapping:
+        - dps_val: disarmed
+          value: disarmed
+        - dps_val: arm
+          value: armed_away
+        - dps_val: home
+          value: armed_home
+        - dps_val: sos
+          value: triggered
+    - id: 1
+      type: string
+      name: trigger
+      mapping:
+        - dps_val: disarmed
+          value: false
+        - dps_val: arm
+          value: false
+        - dps_val: home
+          value: false
+        - dps_val: sos
+          value: true
+        - value_redirect: alarm_state
```

Comment:
Probably this mapping can just be 
```
   - dps_val: sos
     value: true 
   - value: false
```
Since the source code only uses the true value to trigger the alarm, the false value is read only.

But, when triggered is reported by the alarm_state dp, the behaviour will be the same without the explicit trigger dp, so you can leave the whole definition out.

Reply:
The first suggestion works, the second does not. I have adapted it accordingly.

Reply:
Yes, on review the code does not set the feature flag for triggering the alarm if there is no trigger dp, so probably HA will not let you do it.